tensorflow/core/kernels/depthwise_conv_grad_op.cc

```diff
@@ -1055,6 +1059,10 @@ class DepthwiseConv2dNativeBackpropFilterOp : public OpKernel {
     } else {
       LOG(ERROR) << "Only half, float, and double are supported.";
     }
+    // Use CuDNN grouped conv (filter gradients) when input/output is
+    // float16(half). See cudnn release note 7.6.3. (https://docs.nvidia.com/dee
+    // plearning/sdk/cudnn-release-notes/rel_763.html#rel_763)
+    use_cudnn_grouped_conv_ = CUDNN_VERSION >= 7603 && dtype_ == DT_HALF;
```

Comment:
This could be because I'm not familiar with the code I'm reviewing, but the pertinent bit of the release notes says:

 * Improved the performance of grouped convolution, for cudnnConvolutionBackwardFilter() in the configuration below:
   - Algorithm is CUDNN_CONVOLUTION_BWD_FILTER_ALGO_1
   - Math type is CUDNN_TENSOR_OP_MATH or CUDNN_TENSOROP_MATH_ALLOW_CONVERSION
   - Tensor format for filter is NHWC
   - Input and outputs are in FP16 and computation is in FP32
   - On NVIDIA Volta (compute capability 7.0)
   - Per group C & K == 4/8/16/32
   - Filter size: 1x1, 3x3

Here we only check that the input and outputs are fp16.  What about the other preconditions?  Are they checked somewhere else?

Reply:
I think you are referring the grouped conv. But, this PR only focuses on the depthwise conv. For the depwise conv, we check the inputs layout, precision, and stride. See below.
```
Improved depth-wise convolution for forward, dgrad, and wgrad under the following conditions:
Algorithm is algo1
Tensor format for filter is NCHW (wgrad supports NHWC also)
Input and outputs are in FP16 and computation is in FP32
Filter size: 1x1, 3x3, 5x5, 7x7 (dgrad only supports stride 1)
Math type is CUDNN_DEFAULT_MATH
```

Reply:
Thank you, that was helpful.  Should we be checking the filter sizes here?  If not, can you add a comment explaining why?

And as far as the algorithm constraint goes, are you relying on auto-tuning automatically selecting algo1 because it will be fast?  That would be nice to explain in a comment as well.